Bring stability up-to-date for 1.0

eteq opened the pull request:
Prompted by @astrofrog's question in #2758, this does three things with the stability document:

1. Fixes the rendered stability document so that the bullet points come out the correct size (see http://eteq.github.io/astropy/stability.html vs. http://devdocs.astropy.org/stability.html).  This is accomplished in a way that technically violates the HTML spec (CSS ``<style>`` tags inside the ``<body>``), but the internet suggests that basically all web browsers in the last 5-10 years support this anyway.
2. Changes `constants` and `convolution` to blue/stable from yellow/dev
3. A variety of updated descriptions


Two changes I thought about but wasn't sure if I should actually do:  
1. Should `time` be upgraded to green?  It hasn't seen major changes for a while, and I imagine it won't?  @taldcroft and @mhvk, what do you think?
2. Should `nddata` be upgraded to blue?  With APE7 it now has a firm basis so we don't expect to break backwards compatibility now, right?  @astrofrog @mwcraig?

taldcroft commented:
@eteq - I would say `time` is now mature from the stability and API perspective.

taldcroft commented:
And hopefully by 1.1 or 1.2 we can declare Table to be green, but it's probably not there yet.

mhvk commented:
@eteq - agree that `Time` should be stable; to come is barycentric corrections and maybe time-zones, but I think we can guarantee that neither will change the existing API (with a tiny worry about `EarthLocation`, but really it should be possible to make anything backwards compatible).

mhvk commented:
@eteq - could you have the new version displayed somewhere? The usual `view` doesn't work here, and the diff is in this case particularly hard to read since it doesn't really give the context.

mhvk commented:
I think the API for `units` is now stable as well, even if there are still additions to come (logarithmic units, uncertainties, considerable speed-up with `__numpy_ufunc__`).

mhvk commented:
Agreed with @wkerzendorf on another thread that `nddata` should not yet go "reasonably stable" -- we should certainly hope so, but in real life it may well turn out that despite all the thought that got into the APE, the present idea does not work well enough. In particular, we should actually be able to read/write them!

mhvk commented:
I guess I'm slightly confused about the definitions: does "mature" simply suggest that any code that uses that subpackage will continue to work indefinitely, or also that a subpackage will not gain much that is new.  Should there perhaps be two status buttons? One for the API, one for completeness?

taldcroft commented:
I think "mature" just means that the API is very stable so that code using the subpackage is highly unlikely to break.  For instance Table doesn't quite meet that criterion e.g. because of the updates in 1.0 that introduce small API changes.  But I don't see why we need to have a separate button for completeness.  There is no problem for users if a package continues to grow functionality as long as API and performance don't suffer.

mwcraig commented:
@eteq @astrofrog  -- it would be nice to see more extensive use of `nddata` before we mark it as stable. While I don't see the ABC changing much, I would be surprised if we don't find some issues in the current implementation. The description next to it could be changed to:

---
Significantly revised in 1.0. While major changes in the API are not anticipated, broader use may reveal flaws that require API changes.

----

There is a compatibility class in 1.0 that tries to replicate the functionality in 0.4 (though @crawfordsm and I realized last week it is missing a couple of properties) but using it requires making some (hopefully) minimal code changes.

@mhvk @wkerzendorf

eteq commented:
@mhvk - The distinction between "stable" and "mature" is admittedly not that well defined (and perhaps shouldn't be, because they're all different on some level).  To me, "mature" means "hasn't changed too much so likely to have no new bugs, unlikely to have large-scale changes in the future."  While "stable" is "plan is to maintain backwards compatibility, but either large additions are planned or big changes are recent enough that there's a decent change of some significant bugs".  I guess that's pretty similar to what @taldcroft said.

eteq commented:
Alright, I updated the `nddata` as @mwcraig suggested, and set time to mature.  @mhvk - you can see a current version at http://eteq.github.io/astropy/stability.html (I just set this to the latest version).

I think the only think remaining is `vo`. @pllim @mdboom, I didn't quite follow the conclusion from #2758 - currently `vo` is in "dev" status.  Should I change it to "stable"?

pllim commented:
@eteq , sure, Cone Search API is unlikely to change at this point. Let's stabilize it!

eteq commented:
@pllim - done in the latest commit!  

Also, I made some changes to the top part of the stability page.  It referenced some "until 1.0" bits which I just removed and moved the rest of the content to the description of mature/stable/etc.

eteq commented:
@astrofrog - do you want to quick review the description changes and such?  I think this is ready now.

mhvk commented:
@eteq - ok, thanks for the clarification. I agree that for `units` we expect significant additions and thus are not at "mature" -- your text would seem to capture that well.

astrofrog commented:
:+1:

eteq commented:
Alright, I'll go ahead and merge this now.  If anyone else has more changes, fill free to issue another PR as long as it's before the actual release (hopefully no more than a couple days?)
